llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -1708,6 +1710,13 @@ bool SIInsertWaitcnts::insertWaitcntInBlock(MachineFunction &MF,
     }
 
     ++Iter;
+    if (ST->isPreciseMemoryEnabled() && Inst.mayLoadOrStore()) {
+      auto builder =
+          BuildMI(Block, Iter, DebugLoc(), TII->get(AMDGPU::S_WAITCNT))
+              .addImm(0);
```

Comment:
On GFX10+ VMEM stores should have S_WAITCNT_VSCNT 0 as well as (or instead of) this.

Reply:
Done.